### ISSUE TITLE
GitHub version for new projects is wrong

### ISSUE THREAD

**Issue by abitrolly** [release-monitoring/anitya#854]
https://release-monitoring.org/project/49126/ import is full of errors.

1. version includes product name
2. semantic version is selected, but it didn't error out on invalid version
3. versions sorted in reverse order

![Screenshot from 2019-11-27 12-26-35](https://user-images.githubusercontent.com/8781107/69710898-3c2e3980-1111-11ea-8215-e99f0853cb93.png)

**Comment by abitrolly** [comment]
Looks like it might be fixed already https://github.com/release-monitoring/anitya/commit/8d6923e42ec91705d629a1c5950be99b3d0897ef

When it can be deployed?

**Comment by Zlopez** [comment]
You could look at the milestone for [0.18.0](https://github.com/release-monitoring/anitya/milestone/5). I would be glad to have all the remaining issues done before releasing a new version.

**Comment by Zlopez** [comment]
About the reverse ordering, I'm not sure how semver library is working when the version can't be validated. I think it just show the version in the same order as they were received.

**Comment by abitrolly** [comment]
Should it fail with an error when a version doesn't comply?

**Comment by Zlopez** [comment]
This is not a good behavior, it's better if the versions are at least retrieved. What will happen if you change the versioning to RPM?

**Comment by abitrolly** [comment]
Well, maybe add a checkbox to save invalid versions anyway?

**Comment by abitrolly** [comment]
I can not help with new milestone. I tried to put Anitya into container in #805 for easy testing and stuck on provisioning the container by reusing Ansible playbooks.

**Comment by Zlopez** [comment]
I tried to use container a while ago, but without luck. So I stayed with vagrant.

**Comment by Zlopez** [comment]
I was able to tinker with the project https://release-monitoring.org/project/49126/ a little and it's now working as expected.

One thing that is troubling me, is the wrong sort order of the semantic version even when the prefixes are already removed. I tried to create a test for this, but it worked fine. But I will try the same project setup locally and see if I will find the reason for the incorrect sorting.

**Comment by Zlopez** [comment]
I did one small change on the project and it's now working as it should. The issue with sorting was really in prefix removing. The field is stripped of whitespaces before saved to db, so there was one space in the compared versions and thus the version was not considered valid.

I already did the change with using tags instead of release names, when retrieving releases and every other issue on this project was solved by editing the project, so I'm closing this issue.
